cmd/zfs/zfs_main.c

```diff
@@ -4721,8 +4789,15 @@ zfs_do_send(int argc, char **argv)
 	if (flags.replicate && fromname == NULL)
 		flags.doall = B_TRUE;
 
-	err = zfs_send(zhp, fromname, toname, &flags, STDOUT_FILENO, NULL, 0,
-	    flags.verbosity >= 3 ? &dbgnv : NULL);
+	err = zfs_send(zhp, fromname, toname, &flags, STDOUT_FILENO,
+	    exclude_context.count > 0 ? zfs_do_send_exclude : NULL,
+	    &exclude_context, flags.verbosity >= 3 ? &dbgnv : NULL);
+
+	if (exclude_context.count > 0) {
+		for (size_t i = 0; i < exclude_context.count; i++)
+			free(exclude_context.list[i]);
+		free(exclude_context.list);
+	}
```

Comment:
Since we allocate in a separate function, we should do the cleanup in a separate function as well.

Reply:
I thought about that, thought it was too small, and have now done it.